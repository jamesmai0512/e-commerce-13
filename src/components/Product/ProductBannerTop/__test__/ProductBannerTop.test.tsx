import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import ProductBannerTop, { ProductBannerTopProps } from '../'

describe('ProductBannerTop', () => {
  const props: ProductBannerTopProps = {
    title: 'Product Title',
    buttonTheme: 'primary',
    buttonText: 'Buy Now',
    description: 'Product description',
    imageUrl: 'https://example.com/image.jpg',
  }

  it('Should render correctly', () => {
    const component = render(<ProductBannerTop {...props} />)
    expect(component).toMatchSnapshot()
  })
})
