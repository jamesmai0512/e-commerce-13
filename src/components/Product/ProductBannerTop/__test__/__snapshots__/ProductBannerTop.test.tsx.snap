// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductBannerTop Should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="product_banner"
      >
        <div
          class="product_banner_content"
        >
          <div>
            <h2
              class="product_banner_title"
            >
              Product Title
            </h2>
            <button
              class="button primary"
              data-testid="button-component"
            >
              Buy Now
            </button>
          </div>
          <p
            class="product_banner_description"
          >
            Product description
          </p>
        </div>
        <div
          class="product_banner_image"
        >
          <img
            alt="product-banner"
            data-nimg="1"
            decoding="async"
            height="584"
            loading="eager"
            src="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=1080&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=1080&q=75 2x"
            style="color: transparent;"
            width="520"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="product_banner"
    >
      <div
        class="product_banner_content"
      >
        <div>
          <h2
            class="product_banner_title"
          >
            Product Title
          </h2>
          <button
            class="button primary"
            data-testid="button-component"
          >
            Buy Now
          </button>
        </div>
        <p
          class="product_banner_description"
        >
          Product description
        </p>
      </div>
      <div
        class="product_banner_image"
      >
        <img
          alt="product-banner"
          data-nimg="1"
          decoding="async"
          height="584"
          loading="eager"
          src="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=1080&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=1080&q=75 2x"
          style="color: transparent;"
          width="520"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
